ipykernel/iostream.py

```diff
@@ -63,24 +59,33 @@ def __init__(self, socket, pipe=False):
         self.background_socket = BackgroundSocket(self)
         self._master_pid = os.getpid()
         self._pipe_flag = pipe
-        self.io_loop = IOLoop(make_current=False)
+        self.io_loop = None
+        self.loop_ready = threading.Event()
+        self.thread = threading.Thread(target=self._thread_main, name="IOPub")
+        self.thread.daemon = True
+        self.thread.pydev_do_not_trace = True
+        self.thread.is_pydev_daemon_thread = True
+        self.thread.name = "IOPub"
+        self.thread.start()
```

Comment:
It is not a good idea to start a thread in the init (people that want to extend this class will be very annoyed by that). We should respect the behavior of built-in Thread and we should avoid changing the behavior for backward compatible reason.